src/main/java/com/sogeti/SoSenseiPsyche/model/Game.java

```diff
@@ -1,42 +1,75 @@
+import java.util.List;
 import java.util.Scanner;
 
 public class Game {
     public static final int CODE_LENGTH = 4;
+    public static final int MAX_ATTEMPTS = 12;
+    public int attempt = 0;
+    private Code secretCode;
+    private String[] newGuess;
+    private List<String> guesses;
 
 // for testing purpose
 
     public void startGame() {
-        System.out.println("Welcome to Mastermind!");
-        System.out.println("Available colors: ");
+        guesses = new ArrayList<>();
+        secretCode = new Code();
+        System.out.println("Welcome to So Sensei Psyche!");
+        System.out.println("You have 12 attempts to crack the code.");
+        System.out.println("You can choose from the following colors: " + Arrays.toString(Color.values()).replace("[", "").replace("]", ""));
 
-        for (Color color : Color.values()) {
-            System.out.println(color.name() + " (" + color.name().charAt(0) + ")");
-        }
+        secretCode.generateCode();
+
+        // Game logic: make guess, check if game is solved, get feedback, new
+        // guess or game solved or max attempts reached is game over.
+        while (!isGameOver()) {
+            newGuess = makeGuess(new Scanner(System.in));
+            guesses.add(Arrays.toString(newGuess));
+            for (String guess : guesses) {
+                System.out.println(guess.replace("[", "").replace("]", ""));
+            }
 
-        makeGuess(new Scanner(System.in));
+            // feedback logic
+
+            attempt++;
+        }
     }
 
+    // TODO: return type should be Feedback. Adjustments need to be made when
+    //  class Feedback and logic is finished.
     public static String[] makeGuess(Scanner scanner) {
-        System.out.print("Enter your guess (e.g., RGBY): ");
-        String input = scanner.nextLine().toUpperCase().trim();
-        if (input.length() != CODE_LENGTH) {
+        System.out.print("Enter your guess (e.g., R G B Y): ");
+        String[] guess = scanner.nextLine().toUpperCase().split(" ");
+
+        if (guess.length != CODE_LENGTH) {
             return null;
         }
-        String[] guess = new String[CODE_LENGTH];
+
         for (int i = 0; i < CODE_LENGTH; i++) {
-            if (!isCorrectColor(input.charAt(i))) {
+            if (!isCorrectColor(guess[i])) {
                 return null;
             }
-            guess[i] = String.valueOf(input.charAt(i));
+            guess[i] = String.valueOf(guess[i]);
         }
         return guess;
     }
-    public static boolean isCorrectColor(char c) {
+
+    public boolean isGameOver() {
+        if (attempt == MAX_ATTEMPTS) {
+            System.out.println("GAME OVER");
+            return true;
+        }
+        return false;
+    }
+
+    public static boolean isCorrectColor(String c) {
         for (Color color : Color.values()) {
-            if (color.name().charAt(0) == c) {
+            if (color.name().startsWith(c)) {
                 return true;
```

Comment:
startsWith or contains? Now it seems weird that only part of it should match, not entirely.